src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -182,10 +197,6 @@ void Run (DirectoryAssemblyResolver res)
 
 			List<TypeDefinition> allJavaTypes = scanner.GetJavaTypes (allTypemapAssemblies, res);
 
-			// Step 2 - Generate type maps
-			//   Type mappings need to use all the assemblies, always.
-			WriteTypeMappings (allJavaTypes, cache);
```

Comment:
Why move this?  Type mappings should be unrelated to the contents of the `.java` files, and `allJavaTypes` appears to be unchanged, so…. What "hidden"/"non-obvious" bits am I not seeing?

Reply:
Generating marshal methods also potentially rewrites assemblies, thus changing token ids and type maps must absolutely see the final ones.